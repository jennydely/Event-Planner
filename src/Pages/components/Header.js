import React, { useContext } from 'react'
import styled from 'styled-components/macro'
import SelectEvents from './common/SelectEvents'
import EventListFilter from './common/EventListFilter'
import CategoryFilter from './CategoryFilter'
import Searchbar from './Searchbar'
import { UserContext } from "../../providers/UserProvider";

export default function Header({
  onSelectFilter,
  onSelectEventFilter,
  onSelectEventListFilter,
  eventArray,
  eventFilter,
  chosenEventListFilter,
  categoryFilter,
  handleEventSuggestion,
  handleEventSearch,
}) {
  const hasHiddenEvent = eventArray.some((event) => event.isHidden === true)
  const hasOldEvent = eventArray.some(
    (event) => event.eventEndDate < new Date().toJSON().slice(0, 10)
  )
  const sortOptions = ['Sort by', 'Date', 'A-z', 'Z-a']
  if (hasHiddenEvent) sortOptions.push('Hidden')
  if (hasOldEvent) sortOptions.push('Old')

  const chosenEventList = [{ name: 'Public events', value: 'Public' }, { name: 'Favorite events', value: 'Favorite' }, { name: 'Private events', value: 'Private' }]
  const user = useContext(UserContext);
  console.log('headerUser', user)
  return (
    <StyledHeader>
      <FilterContainer>
        <SelectEvents
          id="filter"
          name={sortOptions}
          options={sortOptions}
          onSelectEventFilter={onSelectEventFilter}
          value={eventFilter}
        />
        <Searchbar
          handleEventSuggestion={handleEventSuggestion}
          handleEventSearch={handleEventSearch}
        />
        <CategoryFilter
          onSelectFilter={onSelectFilter}
          categoryFilter={categoryFilter}
        />
      </FilterContainer>
      {   user ? <EventListFilter
        onSelectEventListFilter={onSelectEventListFilter}
        id="EventListfilter"
        name={chosenEventList}
        options={chosenEventList}
        value={chosenEventListFilter} /> : ''}

    </StyledHeader>
  )
}
const StyledHeader = styled.header`
`


const FilterContainer = styled.div`
   max-height: 55px;
   display: flex;
    flex-direction: row;
    align-items:flex-end;
    width: 370px;
    justify-content:space-between;
`