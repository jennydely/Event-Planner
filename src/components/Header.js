import React from 'react'
import styled from 'styled-components/macro'
import CategoryFilter from './CategoryFilter'
import SelectEvents from '../common/SelectEvents'

export default function Header({
  onSelectFilter,
  onSelectEventFilter,
  hasHiddenEvent,
  hasOldEvent,
}) {
  const sortOptions = ['date', 'a-Z', 'Z-a']
  if (hasHiddenEvent) sortOptions.push('Hidden')
  if (hasOldEvent) sortOptions.push('Old')

  return (
    <header>
      <ButtonContainer>
        <SelectEvents
          id="filter"
          name={sortOptions}
          options={sortOptions}
          onSelectEventFilter={onSelectEventFilter}
        />
      </ButtonContainer>
      <CategoryFilter onSelectFilter={onSelectFilter} />
    </header>
  )
}

const ButtonContainer = styled.div`
  margin: 5px;
`
