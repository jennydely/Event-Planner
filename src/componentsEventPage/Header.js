import React from 'react';
import styled from 'styled-components/macro'
import CategoryFilter from './CategoryFilter'

export default function Header({ onSelectFilter }) {
    return (
        <header>
            <ButtonContainer>
                <button name="all" onClick={() => onSelectFilter('all')}>All</button>
            </ButtonContainer>
            <CategoryFilter onSelectFilter={onSelectFilter} />
        </header>
    )
}

const ButtonContainer = styled.div`
   margin: 5px;
   width: wh%;
`