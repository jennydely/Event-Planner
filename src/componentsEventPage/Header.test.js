import React from 'react'
import { render } from '@testing-library/react'
import renderer from 'react-test-renderer'
import Header from './Header'
import 'jest-styled-components'

describe('Header', () => {
    const onSelectFilter = 'all'

    it('displays the filter all button', () => {
        const { getByText } = render(<Header onSelectFilter={onSelectFilter} />)
            expect(getByText('All')).toBeInTheDocument()
    })
    it('renders correctly', () => {
        const tree = renderer.create(<Header onSelectFilter={onSelectFilter} />)
        expect(tree).toMatchSnapshot()
    })
})