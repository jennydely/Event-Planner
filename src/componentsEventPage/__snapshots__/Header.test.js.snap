// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly 1`] = `
.c1 {
  margin: 5px;
  backgroundcolor: grey;
}

.c2 {
  background: rgba(49,42,42,0.75);
}

.c3 {
  background: rgba(67,40,24,0.70);
}

.c4 {
  background: rgba(248,149,17,0.46);
}

.c5 {
  background: rgba(153,88,42,0.70);
}

.c0 {
  margin: 5px;
}

<header>
  <div
    className="c0"
  >
    <button
      name="all"
      onClick={[Function]}
    >
      All
    </button>
  </div>
  <div
    className="c1"
  >
    <button
      className="c2"
      name="metal"
      onClick={[Function]}
    >
      metal
    </button>
    <button
      className="c3"
      name="medieval"
      onClick={[Function]}
    >
      medieval
    </button>
    <button
      className="c4"
      name="sand"
      onClick={[Function]}
    >
      sand
    </button>
    <button
      className="c5"
      name="other"
      onClick={[Function]}
    >
      otters
    </button>
  </div>
</header>
`;
